packages/core/src/serialization.ts

```diff
@@ -321,15 +326,21 @@ export function getExternalReducers(
       const name = global.crypto.randomUUID();
       const type = getStreamType(value);
 
-      const writable = new WorkflowServerWritableStream(name);
-      if (type === 'bytes') {
-        ops.push(value.pipeTo(writable));
-      } else {
-        ops.push(
-          value
-            .pipeThrough(getSerializeStream(getExternalReducers(global, ops)))
-            .pipeTo(writable)
-        );
+      // Only pipe stream data if we have a runId
+      // TODO: Does this cause any issues?
```

Comment:
This is one thing I'm not quite sure about in this PR: because we use `WorkflowServerWritableStream` to finish streams before serialization, we end up skipping this if we don't have a runId, but I'm not sure if this creates any edge cases.
Also unsure if it's worth simply requiring a runId to be passed for reading from streams too (for parity) and then having to pass `runId` to every reducer/reviver all the time, which is typesafe but annoying